mqtt_test.go

```diff
@@ -82,13 +82,14 @@ func TestEncodeDecode(t *testing.T) {
 		{
 			Comment: "CONNACK message",
 			Msg: &ConnAck{
+				SessionPresent:true,
 				ReturnCode: RetCodeBadUsernameOrPassword,
 			},
 			Expected: gbt.InOrder{
 				gbt.Named{"Header byte", gbt.Literal{0x20}},
 				gbt.Named{"Remaining length", gbt.Literal{2}},
 
-				gbt.Named{"Reserved byte", gbt.Literal{0}},
+				gbt.Named{"Reserved byte", gbt.Literal{1}},
```

Comment:
Just a nitty detail: could "Reserved byte" be replaced by "Connect Acknowledge Flags"?

This seems to be the name given here: http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc385349253

Reply:
agree, done and PR is updated.